compiler/src/codegen/comp_utils.re

```diff
@@ -92,23 +92,31 @@ let grain_env_name = "_grainEnv";
 
 let is_grain_env = str => grain_env_name == str;
 
-let get_exported_names = (~local_names=?, wasm_mod) => {
+let get_exported_names = (~function_names=?, ~global_names=?, wasm_mod) => {
   let num_exports = Export.get_num_exports(wasm_mod);
   let exported_names: Hashtbl.t(string, string) = Hashtbl.create(10);
   for (i in 0 to num_exports - 1) {
     let export = Export.get_export_by_index(wasm_mod, i);
     let export_kind = Export.export_get_kind(export);
-    if (export_kind == Export.external_function
-        || export_kind == Export.external_global) {
-      let exported_name = Export.get_name(export);
-      let internal_name = Export.get_value(export);
+    let exported_name = Export.get_name(export);
+    let internal_name = Export.get_value(export);
+
+    if (export_kind == Export.external_function) {
+      let new_internal_name =
+        switch (function_names) {
+        | Some(function_names) => Hashtbl.find(function_names, internal_name)
```

Comment:
This `Hashtbl.find` will never raise?

Reply:
It would only raise if the linker failed to provide the names of all the functions/locals, but if that happened the linker would fail far before it ever got here.